scripts/word_freq_table.js

```diff
@@ -0,0 +1,50 @@
+window.addEventListener("DOMContentLoaded", function () {
+    function checkAllVisible() {
+        const checkBoxes = document.getElementsByClassName(`cb_${this.dataset.instance}`);
+        for (const checkBox of checkBoxes) {
+            const tBody = checkBox.closest("tbody");
+            if (tBody.style.display !== "none") {
```

Comment:
Note that `style` returns a `CSSStyleDeclaration` on Chrome (the same return as `getComputedStyle`) so this may still trigger a synchronous style resolution. The reason is that to answer this comparison, the browser has to run the  CSS cascade to know which style applies to the element.

The way around this is to add some "invisible" class. Classes are set in the DOM and don't cascade so they can be queried without forcing a style resolution.

TBH, our pages are fairly static (no animation/transition that would quickly updates the style) so it's not clear if we should be concerned about any synchronous style resolution.

Reply:
The way it is working seems fast enough in the examples I've tried but there could be thousands of checkboxes with only one word frequency which the user did not want to select so it could be better to first select the tbody elements and then for any that are vissible check its child checkboxes. I'll try that out.